src/pages/admin/ReceivePayment.vue

```diff
@@ -40,27 +40,40 @@ export default {
   components: {QRCodeScanner},
   computed: {
     ...mapGetters('admin', ['getIsDialogOpenned', 'getReceived', 'getIsDisabled']),
+    isDisabled() {
+      let reg = /^[1-9]+(\.?\d{1,})?$/
+      return !reg.test(this.amount) || this.getIsDisabled
+    },
   },
   methods: {
     ...mapActions('admin', ['receivePayment']),
-    ...mapMutations('admin', ['setIsDialogOpenned']),
+    ...mapMutations('admin', ['setIsDialogOpenned', 'setIsDisabled']),
     receiveScannerCode(jwtPayment) {
       this.closeScanner()
       this.receivePayment({amount: this.amount, jwtPayment})
     },
     closeScanner() {
       this.scannerStarted = false
       this.theScanner = null
+      this.setIsDisabled(false)
     },
     openScaner() {
       this.scannerStarted = true
       this.theScanner = QRCodeScanner
+      this.setIsDisabled(true)
     },
     closeDialog() {
       this.setIsDialogOpenned(false)
       this.amount = ''
     },
   },
+  mounted() {
+    this.$el.querySelector('input').addEventListener('keypress', evt => {
```

Comment:
you write code like this, coz it's very hard to maintain code written like this
should always wrap a piece of code in a function, mounted function is triggered at mounted, that's the meaning of it
--> if (charCode > 31 && (charCode < 48 || charCode > 57) && charCode !== 46) evt.preventDefault()
      else return true  --> what does this means, no one could know